DragaliaAPI/DragaliaAPI/Features/StorySkip/StorySkipController.cs

```diff
@@ -0,0 +1,88 @@
+using DragaliaAPI.Controllers;
+using DragaliaAPI.Features.Quest;
+using DragaliaAPI.Models.Generated;
+using DragaliaAPI.Services;
+using DragaliaAPI.Shared.MasterAsset;
+using DragaliaAPI.Shared.MasterAsset.Models;
+using DragaliaAPI.Shared.MasterAsset.Models.Story;
+using DragaliaAPI.Shared.PlayerDetails;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DragaliaAPI.Features.StorySkip;
+
+[Route("story_skip")]
+public class StorySkipController : DragaliaControllerBase
+{
+    private readonly ILogger<StorySkipController> logger;
+    private readonly IPlayerIdentityService playerIdentityService;
+    private readonly IStorySkipService storySkipService;
+    private readonly IUpdateDataService updateDataService;
+
+    public StorySkipController(
+        ILogger<StorySkipController> logger,
+        IPlayerIdentityService playerIdentityService,
+        IStorySkipService storySkipService,
+        IUpdateDataService updateDataService
+    )
+    {
+        this.logger = logger;
+        this.playerIdentityService = playerIdentityService;
+        this.storySkipService = storySkipService;
+        this.updateDataService = updateDataService;
+    }
+
+    [HttpPost("skip")]
+    public async Task<DragaliaResult> Read(
+        CancellationToken cancellationToken
+    )
+    {
+        try
+        {
+            this.logger.LogDebug("Beginning story skip for player.");
+
+            IEnumerable<QuestData> questDatas = MasterAsset.QuestData.Enumerable.Where(x =>
+                x.Gid < 10011 && x.Id > 100000000 && x.Id.ToString().Substring(6, 1) == "1"
```

Comment:
Why do you need the filter on `Id > 100000000` and that the 6th digit is `1`?

Reply:
While comparing the different quest Ids, I found that index 6 was a flag for the easy mode missions. I tested this out, and it worked nicely.
1 = easy, 2 = hard, 3 = very hard
The chapter number and quest number are encoded in there as well.

Because I was doing the substring, I was hitting a string length error on that Id. Instead of investigating further, I filtered for Id > 100000000.